docs:add llgo get usage detail (#4)
seri037/llcppg: llpkgstore.md

**Maintenance:**
- Updates must be published as new versions with -patchX suffix
- Example: `version` -> `version-patch1`

> ## Llgo Get Usage

Reviewer: ```suggestion
## LLGo Get Usage
```

Install LLpkg using:
```bash
llgo get clib@cversion
llgo get module_path@module_version
```

1. `clib` corresponds to the original library name in C, and `cversion` corresponds to the original version in C.
2. `llgo get clib`: the module path of the C library needs to be queried from the version mapping table based on clib, which is equivalent to `llgo get module_path@latest` and retrieves the latest llpkg's Go module of clib.
3. Suggestion: Add `llgo clist clib` to inform users of the version mapping information of clib(mapping info is generated based on llpkgstore.json).
  ```
  The available Go module version of this C original version:
  format: c original version => go module versions

>   1.3 => ["v0.1.0", "v1.0.1"]

Reviewer 2: It's better to specify how llgo selects a version.

  1.3.1 => ["v1.1.0"]
  ```

Install LLpkg process:

1. `llgo get` follows the Go Module to obtain the corresponding version of the Go module.
2. Check if the pulled Go module contains `llpkg.cfg` to determine if it is a llpkg.

> 3. If it is, `llgo get` need do extra things: 

Reviewer: ```suggestion
3. If it is, `llgo get` needs to do extra things: 
```

>   - Run `conan install` for binaries and Store `.pc` files for building In LLGOMODCACHE which is an env variable.

Reviewer: ```suggestion
  - Run `conan install` for binaries and store `.pc` files for building in `${LLGOMODCACHE}`.
```

>   - Add a comment in the `go.mod` file indicating the original version of the corresponding C library for this llpkg and Ignore indirect dependence for now.

Reviewer: ```suggestion
  - Add a comment in the `go.mod` file indicating the original version of the corresponding C library for this llpkg and ignore indirect dependence for now.
```

  ```go.mod
  require (

>       github.com/google/llpkg/cjson v1.1.0  // cjson-1.7.18

Reviewer 2: What about indirect module?

  )
  ```

> ## LLGOMODCACHE Env Variable Design

Reviewer 2: It should be LLGOMODCACHE Directory Design

> One use is to store the binary `.pc` files of the C library and allow `llgo build` to find them

Reviewer: ```suggestion
One usage is to store `.pc` files of the C library and allow `llgo build` to find them
```

> 1. official path is `{GOPATH}/llgo/pkg/mod`

Reviewer: ```suggestion
1. Initialized as `${GOPATH}/llgo/pkg/mod`
```

> 2. if GOPATH is empty, it defaults to `{HOME}/llgo/pkg/mod`

Reviewer: ```suggestion
2. if GOPATH is empty, it defaults to `${HOME}/llgo/pkg/mod`
```

> 3. `{LLGOMODCACHE}/{module_path}/{module_name}@{module_version}/` stores a binary `.pc` file of the C library corresponding to the Go Module

Reviewer 2: What's the relationship between `LLGOMODCACHE` and `GOPATH`? Need more detail.
